File: editor/en/[2444]Count Subarrays With Fixed Bounds.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
    long count = 0L;
    int lastMinPosition = -1;
    int lastMaxPosition = -1;
    int lastInvalidPosition = -1;

    for (int i = 0; i < nums.length; i++) {
      int v = nums[i];

      if (v < minK || v > maxK) {
        lastInvalidPosition = i;
            }
      if (v == minK) {
        lastMinPosition = i;
            }
      if (v == maxK) {
        lastMaxPosition = i;
      }

      int validStartCount = Math.min(lastMinPosition, lastMaxPosition) - lastInvalidPosition;
      if (validStartCount > 0) {
        count += validStartCount;
      }
    }
    return count;
  }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2444]Count Subarrays With Fixed Bounds.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        count = 0
        last_min = -1
        last_max = -1
        last_invalid = -1

        for i, num in enumerate(nums):
            if num < minK or num > maxK:
                last_min = i
                last_max = i
                last_invalid = i
            if num == minK:
                last_min = i
            if num == maxK:
                last_max = i
            count += min(last_min, last_max) - last_invalid

        return count
# leetcode submit region end(Prohibit modification and deletion)
